src/main/java/org/cactoos/iterator/IteratorChar.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.cactoos.iterator;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Wrap array of character into {@link Iterator}.
+ *
+ * @author Alexander Menshikov (sharplermc@gmail.com)
+ * @version $Id$
+ * @since 0.30
+ */
+public final class IteratorChar implements Iterator<Character> {
```

Comment:
This class duplicates functionality of [IteratorOf](https://github.com/yegor256/cactoos/blob/63f3b7fcd7be411d573c8ef3ac99bdb10fa4b144/src/main/java/org/cactoos/iterator/IteratorOf.java) and [IterableOf](https://github.com/yegor256/cactoos/blob/63f3b7fcd7be411d573c8ef3ac99bdb10fa4b144/src/main/java/org/cactoos/iterable/IterableOf.java). Please, remove it.

Reply:
I'm sorry, but you are not right.
```Java
Iterator<Character> iter0 = new IteratorOf<>("foobar".toCharArray()); // compile error
Iterator<Character> iter1 = new IteratorChar("foobar".toCharArray()); // okay
```
Because code `new IteratorOf<>("foobar".toCharArray())` typed into `new IteratorOf<char[]>("foobar".toCharArray())`. It's how generic works in Java.

Maybe I missed the elegant way in `cactoos` how to convert `char[]` into `Character[]`.

In Apache commons there is [`ArrayUtils.toObject`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/ArrayUtils.html) for such conversion.

Reply:
@SharplEr Yes! You're absolutely right, thank you! I didn't check my examples, I just assumed that they would work, because that seemed logical to me. Please, next time, if you disagree with the reviewer, disagree as soon as possible before trashing the code completely.

My main concern regarding the initial code was [the static utility procedure](https://github.com/yegor256/cactoos/pull/729#discussion_r175607549). That's it. I don't actually care how you get rid of it. I [suggested](https://github.com/yegor256/cactoos/pull/729#discussion_r175946667) to create an object, because [Each Private Static Method Is a Candidate for a New Class](http://www.yegor256.com/2017/02/07/private-method-is-new-class.html), you said it's slow, so I suggested something else. I'm just suggesting you your options. I'm not the `DEV`, it's up to you how you resolve the [identified problem](https://github.com/yegor256/cactoos/pull/729#discussion_r175607549) (no, removing `static` keyword is not the solution).

Just keep in mind that the more code you write, the more problems you introduce. I would gladly accept even your initial code, if it didn't have that utility procedure in the body of the object.

Reply:
@rexim I'm just trying to find the best solution. And your argument about `IteratorOf` is wrong, as your example. So I just fixed your example to reach the goal. And this way turned out to be wrong somehow... Also, Cactoos has a rule for avoiding object creation in a loop (`AvoidInstantiatingObjectsInLoops`), so all your remarks was a bit confusing.